src/NLog/IEventProperties.cs

```diff
@@ -0,0 +1,63 @@
+﻿// 
+// Copyright (c) 2004-2016 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Extends Dictionary to have ordered subset of items
+    /// </summary>
+    public interface IEventProperties : IDictionary<object, object>
+    {
+        /// <summary>
+        /// Add propery to the end of the collection 
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        bool AddOrderedProperty(string key, object value);
+
+        /// <summary>
+        /// Remove ordered property by key
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns>found?</returns>
+        bool RemoveOrderedProperty(string key);
+
+        /// <summary>
+        /// Get ordered properties in order of the template
+        /// </summary>
+        /// <returns></returns>
+        IEnumerable<KeyValuePair<string, object>> OrderedProperties { get; }
```

Comment:
I'm doubting of ` IList<KeyValuePair<string, object>>` instead of ` IEnumerable<KeyValuePair<string, object>>`.  What do you think is best? (note that it's a getter only)

Reply:
Well the IList-interface have an Add-method. Not sure I want to support that.